customfuncs/javascript.go

```diff
@@ -60,32 +61,45 @@ func parseArgTypeAndValue(argDecl, argValue string) (name string, value interfac
 		}
 		return name, b, nil
 	default:
-		return "", nil, fmt.Errorf("arg_type '%s' in '<arg_name>:<arg_type>' is not supported", declParts[1])
+		return "", nil, fmt.Errorf("arg_type '%s' in '<arg_name>:<arg_type>' is not supported", argDecl[col+1:])
 	}
 }
 
+// JSProgramCache caches *goja.Program. A *goja.Program is compiled javascript and it can be used
+// across multiple goroutines and across different *goja.Runtime. If default loading cache capacity
+// is not desirable, change JSProgramCache to a loading cache with a different capacity at package
+// init time. Be mindful this will be shared across all use cases inside your process.
+var JSProgramCache *caches.LoadingCache
+
+// jsRuntimePool caches *goja.Runtime whose creation is expensive such that we want to have a pool
+// of them to amortize the initialization cost. However, a *goja.Runtime cannot be used by two/more
+// javascript's at the same time, thus the use of sync.Pool. Not user customizable.
+// var jsRuntimePool *resPool
+var jsRuntimePool sync.Pool
```

Comment:
Why we have this naming style differences between `js` and `JS`

Reply:
JS means it's exported/public. js means it's private.